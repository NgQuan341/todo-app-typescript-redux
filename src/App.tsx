import { useEffect } from 'react'
import { useDispatch } from 'react-redux'
import './App.css'
import { fetchTodos } from './components/TodoList/reducer'
import TodoContainer from './page'

function App() {
  const dispatch = useDispatch()
  useEffect(()=>{
    dispatch<any>(fetchTodos())
  },[])
  return (
    <TodoContainer />
  )
}

export default App
