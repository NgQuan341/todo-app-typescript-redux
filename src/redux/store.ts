// import {createStore} from 'redux'
// import rootReducer from './reducer';
// import { composeWithDevTools } from 'redux-devtools-extension'

// const composedEnheancers = composeWithDevTools()
// const store = createStore(rootReducer, composedEnheancers);

// export default store

import { configureStore } from '@reduxjs/toolkit'
import todoListSlice from '../components/TodoList/reducer';

const store = configureStore({
    reducer: todoListSlice.reducer
})
export type RootState = ReturnType<typeof store.getState>

export default store