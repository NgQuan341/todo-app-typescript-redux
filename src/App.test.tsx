import React from 'react';
import { shallow } from 'enzyme'
import App from './App';
import { Provider } from 'react-redux';
import store from './redux/store';

const setup = () => {
  let wrapper = shallow<typeof App>(<Provider store={store}><App/></Provider>)
  return wrapper
  
}
describe('App component', () =>{
  test('renders without error', () => {
      const wrapper = setup()                 
      expect(wrapper.html()).toMatchSnapshot()
  })
})