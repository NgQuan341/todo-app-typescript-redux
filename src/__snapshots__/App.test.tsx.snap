// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders without error 1`] = `"<div style=\\"width:500px;margin:10vh auto;display:flex;flex-direction:column;background-color:white;padding:20px;box-shadow:0 0 10px 4px #bfbfbf;border-radius:5px;height:80vh\\"><h2 class=\\"ant-typography\\" style=\\"text-align:center\\">TODO APP</h2><div class=\\"ant-divider ant-divider-horizontal\\" role=\\"separator\\"></div><div class=\\"ant-row\\" style=\\"height:calc(100% - 130px)\\"><div style=\\"height:calc(100% - 40px);overflow-y:scroll\\" class=\\"ant-col ant-col-24\\"></div><div class=\\"ant-col ant-col-24\\"><span class=\\"ant-input-group ant-input-group-compact\\" style=\\"display:flex;margin-top:20px\\"><input type=\\"text\\" value=\\"\\" name=\\"input-todo\\" class=\\"ant-input\\"/><button name=\\"submit-button\\" style=\\"background:#2DEB56;border-color:#2DEB56;color:white\\" type=\\"button\\" class=\\"ant-btn ant-btn-primary button-submit\\"><span>Add</span></button></span></div></div></div>"`;
